lib/posix/options/pthread.c

```diff
@@ -1114,6 +1115,44 @@ int pthread_join(pthread_t pthread, void **status)
 	return 0;
 }
 
+/**
+ * @brief Await a thread termination with timeout.
+ *
+ * Non-portable GNU extension of IEEE 1003.1
+ */
+int pthread_timedjoin_np(pthread_t pthread, void **status, const struct timespec *abstime)
+{
+	if (abstime == NULL) {
+		return EINVAL;
+	}
+
+	if (abstime->tv_sec < 0 || abstime->tv_nsec < 0 || abstime->tv_nsec >= NSEC_PER_SEC) {
+		return EINVAL;
+	}
+
+	return pthread_timedjoin_internal(pthread, status, K_MSEC(timespec_to_timeoutms(abstime)));
```

Comment:
It looks like this absolute timespec should be converted to a relative delay. Before the call to timedjoin_internal, maybe subtract `k_uptime_get()` from the milliseconds value?

Reply:
I thought that was the purpose of `timespec_to_timeoutms`. It subtracts `current time` internally.

Reply:
Yes, sorry - my bad.

Reply:
btw: This gives us essentially only msec resolution. Not sure, if a problem or not.

Reply:
I guess it should be fine, whoever needs finer resolution can probably submit a patch later.